rush-plugins/rush-amazon-s3-build-cache-plugin/src/AmazonS3BuildCacheProvider.ts

```diff
@@ -27,10 +31,7 @@ export class AmazonS3BuildCacheProvider implements ICloudBuildCacheProvider {
   private _rushSession: RushSession;
 
   public get isCacheWriteAllowed(): boolean {
-    return (
-      this._rushSession.EnvironmentConfiguration.buildCacheWriteAllowed ??
-      this._isCacheWriteAllowedByConfiguration
-    );
+    return EnvironmentConfiguration.buildCacheWriteAllowed ?? this._isCacheWriteAllowedByConfiguration;
```

Comment:
Checking if the global environment variable is set to disable cache writes should be done at a higher level than the specific cache provider, since the environment variable doesn't specify the cache provider. It's useful to be able to block cache writes for the local cache as well.

Reply:
Out of scope for this PR. Handled in !3044